packages/studio-ui/tests/components/UndoRedo.test.tsx

```diff
@@ -46,6 +46,24 @@ describe("Undo/redo", () => {
     expect(useStudioStore.getState().pages.activeComponentUUID).toBeUndefined();
   });
 
+  it("undoes last state update using control + z", async () => {
+    render(<UndoRedo />);
+    expect(useStudioStore.getState().pages.activeComponentUUID).toBe(
+      "searchbar-uuid"
+    );
+    await userEvent.keyboard('{Control>}z{/Control}')
+    expect(useStudioStore.getState().pages.activeComponentUUID).toBeUndefined();
+  });
+
+  it("undoes last state update using command + z", async () => {
+    render(<UndoRedo />);
+    expect(useStudioStore.getState().pages.activeComponentUUID).toBe(
+      "searchbar-uuid"
+    );
+    await userEvent.keyboard('{Meta>}z{/Meta}')
+    expect(useStudioStore.getState().pages.activeComponentUUID).toBeUndefined();
```

Comment:
Note that since the reference to `undo` does not change, this `useEffect` will only be run on first render.  Furthermore (fun fact), if you call `addEventListener` with a named function, it will not be added more than once.  So, if there was a case that the `useEffect` hook would be called twice, it would not add a duplicate listener.